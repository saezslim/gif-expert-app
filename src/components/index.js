export * from './AddCategory'
export * from './GifGrid'
export * from './GifItem'
export * from './LoadingScreen'
export * from './FixedHeader'