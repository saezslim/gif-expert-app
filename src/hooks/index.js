export * from './useFetchGifs'
export * from './useFunCopy'