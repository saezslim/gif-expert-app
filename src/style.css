*, *::before, *::after{
  box-sizing: border-box;
}
*{
  margin: 0;
  padding: 0;
  font:inherit;
}

body{
  min-height: 100vh;
}

html{
  color-scheme: dark light;
}

img, svg, picture, video{
  display: block;
  max-width: 100%;
}
h1{
  font-size: 2.25rem;
}
h2{
  font-size: 1.75rem;
}
h3{
  font-size: 1.45rem;
}
h4{
  font-size: 1.15rem;
}
.header{
  display: flex;
  flex-direction: column;
  gap: 1rem;
  text-align: center;
  padding: 1.125em 0;
}
.header__title{
  letter-spacing: .125rem;
  font-weight: bold;
}
.inputSearch{
  width: 100%;
  padding: .45rem .75rem;
  text-align: center;
}
.grid-card{
  display: grid;
  /*grid-template-columns: repeat(5, 1fr);
  gap: 1.25rem;*/
  grid-template-columns: repeat(2, 1fr);
  gap: 1rem;
  padding-inline: .65em;
}
.grid-card__title{
  grid-column: 1/-1;
  text-align: center;
  padding: 1.25em;
  font-weight: bolder;
  letter-spacing: .25rem;
}
@media (min-width: 768px) { 
  .grid-card{
    grid-template-columns: repeat(3, 1fr);
  }
}
@media (min-width: 1024px) { 
  .grid-card{
    grid-template-columns: repeat(5, 1fr);
  }
}
.card{
  display: flex;
  position: relative;
  flex-direction: column-reverse;
  text-align: center;
  align-items: center;
  justify-content: center;
  max-width: 20rem;
  max-height: 20rem;
  box-shadow: none;
}
.card:hover{
  box-shadow: 0px 0px 8px 0px #000000c4;
  transition: box-shadow 350mx ease-in-out;
}
.card::after{
  content: 'Click to copy URL';
  position: absolute;
  top: 0;
  width: 100%;
  padding: .25em 0;
  background-color: rgba(0, 0, 0, .7);
  color: white;
  opacity: 0;
}
.card:where(.card:hover)::after{
  opacity: 1;
  transition:opacity 350ms ease-in-out;

}
.card--click{
  position: relative;
}
.card--click::after{
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 15%;
  content: 'ðŸ˜‰ðŸ‘Œ';
  font-size: 2rem;
  border-radius: 50%;
  width: 6rem;
  height: 6rem;
  background-color: hsl(113, 57%, 53%);
  transform: scale(1);
  animation: emojis .35s ease-in-out infinite;
}

.card__title{
  display: flex;
  font-size: .75rem;
  padding: .75em;
  background-color: #ccc;
  color: black;
  font-weight: bold;
  min-height: 2.75rem;
  align-items: center;
  justify-content: center;
  text-transform: capitalize;
  width: 100%;
}
.card__img{
  aspect-ratio: 4/3;
  object-fit: cover;
}
@keyframes emojis {

  25%{
    transform: scale(.5);
  }
  50%{
    transform: scale(1.5);
  }
}