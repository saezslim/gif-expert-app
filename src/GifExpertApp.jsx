import { useState } from "react"
import { GifGrid, FixedHeader } from "./components"

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Dragon Ball'])
  
  const onAddCategory = (category) => {
    setCategories([category,...categories])
  }
  return(
    <>
      <FixedHeader
        onNewCategory={onAddCategory}
      />
      
        {
          categories.map((category) => (// Si le agregamos el id, renderiza por completo todo
            <GifGrid key={category}
             category={category} 
             />
          ))
        }
    </>
  )
}