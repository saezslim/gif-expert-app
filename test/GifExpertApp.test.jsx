import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Testing <GifExpertApp />', () => {
  test('Snapshot should be the same', () => {
    const {container} = render(<GifExpertApp />)
    screen.debug()
    expect(container).toMatchSnapshot()
  })
  test('Should have an initial value', () => {
    const inputVal = 'Goku'
    render(<GifExpertApp />)
  })
})