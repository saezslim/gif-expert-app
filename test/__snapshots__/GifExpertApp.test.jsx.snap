// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> Should have an initial value 1`] = `
<div>
  <header
    class="header"
  >
    <h1
      class="header__title"
    >
      Gif Expert App
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="inputSearch"
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
  </header>
  <div
    class="grid-card"
  >
    <h3
      aria-label="cardTitle"
      class="grid-card__title"
    >
      Dragon Ball
    </h3>
    <p>
      Cargando...
    </p>
  </div>
</div>
`;

exports[`Testing <GifExpertApp /> Snapshot should be the same 1`] = `
<div>
  <header
    class="header"
  >
    <h1
      class="header__title"
    >
      Gif Expert App
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="inputSearch"
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
  </header>
  <div
    class="grid-card"
  >
    <h3
      aria-label="cardTitle"
      class="grid-card__title"
    >
      Dragon Ball
    </h3>
    <p>
      Cargando...
    </p>
  </div>
</div>
`;
